### DISCUSSION TITLE
Why I can't find Fl_init_all() function in FLTK c code ?

### DISCUSSION THREAD

**Discussion by lbk861** [fltk-rs/fltk-rs#1551] [Q&A] [answered]
I jump the definition which is a binding. But I can't find Fl_init_all() function in FLTK c code.
```
impl Default for App {
    fn default() -> Self {
        init_all();
        App {}
    }
}
```

I want to know how Fl_init_all() works.

**Comment by MoAlyousef** [comment, marked as answer]
Here:
https://github.com/MoAlyousef/cfltk/blob/acac9c805681b4af02b4104cd91f3eb04bec1a44/src/cfl.cpp#L459

It’s unlikely that an lsp or editor would let you jump to a definition in a different language.

**Reply by lbk861** [reply]
> Here: https://github.com/MoAlyousef/cfltk/blob/acac9c805681b4af02b4104cd91f3eb04bec1a44/src/cfl.cpp#L459
> 
> It’s unlikely that an lsp or editor would let you jump to a definition in a different language.

I see. And why fltk-rs generates bindings to cfltk rather than fltk directly ?

**Reply by MoAlyousef** [reply]
The Rust<->C++ interop story isn’t great. There is no automated way to do it. 
Rust-Bindgen breaks with constructors, complex or templated types. cxx requires writing so much boilerplate. autocxx also breaks with constructors and complex types. 
An advantage of creating C bindings is that they can be used with other languages as well. There are zig bindings as well which use cfltk. And of course it can be used from C.

**Reply by lbk861** [reply]
Thanks for your answer!
